### PR TITLE
Pll errorcode

### PR DESCRIPTION
**Pull request by connorl309**
Makes PLL not inf loop the BPS if it doesn't engage, will retry a set amount of times before faulting.

# Quality Assurance Checklist 
To make reviews more efficient, please make sure the software feature meets the following standards and check everything off that meets the quality check. Once everything has been checked, the assigned reviewers will begin the review process. _Edit this description to check off the list._

There are exceptions with all guidelines. As long as your decisions are justified, then you are good! Contact the reviewers or the leads about any exceptions. 

## Requirements 
- [x] Followed Coding Style Guide 
- [x] Code Build checks pass 
- [x] No merge conflicts 
- [x] Software feature has test associated with it. 
- [x] Software feature has documentation for it updated.
- [x] If applicable, does software feature have simulation test associated with it (.json & out.json file)
- [x] Test provides useful information and uses relevant data to accurately represent BPS 
    - NOTE: If test file already exists, use that one 
- [ ] 2 Members of the team have already reviewed my PR 
    - NOTE: Approver will not look at PR if 2 other members haven't reviewed first 
- [x] Did you Have Fun?

## Things to Consider 
- [ ] Even if the above are checked, is this the best way of writing my code? 
    - It's possible to write code that works, but are there ways to make code more efficient?

### CODE REVIEW

File: BSP/STM32F413/Src/BSP_PLL.c [hunk 1]
```diff
@@ -23,40 +28,49 @@ void BSP_PLL_Init(void) {
...
-		FLASH_SetLatency(FLASH_Latency_3);
-		FLASH_PrefetchBufferCmd(ENABLE);
-		FLASH_InstructionCacheCmd(ENABLE);
-		FLASH_DataCacheCmd(ENABLE);
-		
-		// HCLK = SYSCLK
-		RCC_HCLKConfig(RCC_SYSCLK_Div1);
+	// If PLL fails to engage in time retry PLL_ATTEMPTS_MAX amount of times
+	// If still fail then we will fault
+	while (status == ERROR) {
```
Comment (manthanand) [inline BSP/STM32F413/Src/BSP_PLL.c:35]: This crosses layers, you should return error and error handle elsewhere.

File: BSP/STM32F413/Src/BSP_PLL.c [hunk 2]
```diff
@@ -23,40 +28,49 @@ void BSP_PLL_Init(void) {
...
+	// Select the main PLL as system clock source
+	RCC_SYSCLKConfig(RCC_SYSCLKSource_PLLCLK);
 		
-		// Update the system clock variable
-		SystemCoreClockUpdate();
-	}else{
-		while(1);	// Spin for error
-	}
+	// Update the system clock variable
+	SystemCoreClockUpdate();
```
Comment (manthanand) [inline BSP/STM32F413/Src/BSP_PLL.c:72]: Make sure you are using spaces instead of tabs.

File: Apps/Src/main.c [hunk 3]
```diff
@@ -32,7 +32,12 @@ int main() {
     
     OS_ERR err;
 
-    BSP_PLL_Init();
+    if(BSP_PLL_Init() == ERROR){
+        // if initializing PLL fails, something is very wrong
+        Fault_BitMap |= Fault_CRC;
```
Comment (manthanand) [inline Apps/Src/main.c:37]: Make it a Fault_OS. CRC is for SPI.

File: BSP/STM32F413/Src/BSP_PLL.c [hunk 4]
```diff
@@ -14,49 +14,62 @@
...
+	// If PLL fails to engage in time retry PLL_ATTEMPTS_MAX amount of times
+	// If still fail then we will fault
+	while (status == ERROR) {
+		if (pllAttempts >= PLL_ATTEMPTS_MAX) {
+			//return a error for waiting too long
+			return ERROR;
+		}
+		pllAttempts++;
+		status = RCC_WaitForHSEStartUp();
+	}
```
Comment (manthanand) [inline BSP/STM32F413/Src/BSP_PLL.c:41]: this could be shorter, just get rid of the status variable.

File: BSP/STM32F413/Src/BSP_PLL.c [hunk 5]
```diff
@@ -14,49 +14,61 @@
...
     RCC_HSEConfig(RCC_HSE_ON);
-	
-	ErrorStatus status = RCC_WaitForHSEStartUp();
+    
+    ErrorStatus status = RCC_WaitForHSEStartUp();
+    uint8_t pllAttempts = 0;
+
+    // If PLL fails to engage in time retry PLL_ATTEMPTS_MAX amount of times
+    // If still fail then we will fault
+    while (status == ERROR) {
```
Comment (tiandahuang) [inline BSP/STM32F413/Src/BSP_PLL.c:34]: this loop isn't actually updating the `status` variable

File: Tests/Test_BSP_PLL.c [hunk 6]
```diff
@@ -16,7 +16,18 @@ int main(void){
         }
         time--;
     }                           //delay is to show the clock going from 16MHz to 80MHz once BSP_PLL_Init() is called
-    BSP_PLL_Init();
+
+    if(BSP_PLL_Init() == ERROR){
+        // if initializing PLL fails, something is very wrong
+        while(1){
+            printf("PLL init has failed");
+            volatile uint16_t horse = -1;
```
Comment (tiandahuang) [inline Tests/Test_BSP_PLL.c:24]: this is the funniest loop i've ever seen (it's in a test so it's ok)

**Review by tiandahuang — COMMENTED** [review]
looks good (except for the one thing)
make sure you rebase and that everything compiles with no errors & warnings

**Pull request closed without merging**